ci-operator/step-registry/storage/create/csi-aws-efs/cross-account/storage-create-csi-aws-efs-cross-account-commands.sh

```diff
@@ -0,0 +1,315 @@
+PEER_REQUEST_ID=$(aws ec2 create-vpc-peering-connection \
+  --vpc-id "${AWS_ACCOUNT_B_VPC_ID}" \
+  --peer-vpc-id "${AWS_ACCOUNT_A_VPC_ID}" \
+  --peer-owner-id "${AWS_ACCOUNT_A_ID}" \
+  --output json | jq -r '.VpcPeeringConnection.VpcPeeringConnectionId')
+echo "${PEER_REQUEST_ID}" > "$SHARED_DIR"/vpc_peering_id
+
+switch_aws_credentials default
+aws ec2 accept-vpc-peering-connection --vpc-peering-connection-id "${PEER_REQUEST_ID}"
+SUBNETS=$(aws ec2 describe-subnets \
+  --filters Name=vpc-id,Values="${AWS_ACCOUNT_A_VPC_ID}" \
+  | jq -r '.Subnets[].SubnetId')
+
+for subnet in $SUBNETS; do
+  # Get route table associated with this subnet (specific or main)
+  ROUTE_TABLE_ID=$(aws ec2 describe-route-tables \
+    --filters Name=association.subnet-id,Values="$subnet" \
+    | jq -r '.RouteTables[0].RouteTableId')
+
+  if [ "$ROUTE_TABLE_ID" == "null" ]; then
+    # No subnet-specific route table, fall back to main route table
+    ROUTE_TABLE_ID=$(aws ec2 describe-route-tables \
+      --filters Name=vpc-id,Values="${AWS_ACCOUNT_A_VPC_ID}" Name=association.main,Values=true \
+      | jq -r '.RouteTables[0].RouteTableId')
+  fi
+
+  # Check if the route table has a route to an Internet Gateway
+  HAS_IGW=$(aws ec2 describe-route-tables --route-table-ids "${ROUTE_TABLE_ID}" \
+    | jq -e '.RouteTables[0].Routes[]? | select((.GatewayId | type == "string") and (.GatewayId | startswith("igw-")))' \
+  > /dev/null && echo "yes" || echo "no")
+
+  # If no IGW, it's private
+  if [ "$HAS_IGW" == "no" ]; then
+    aws ec2 create-route --route-table-id "${ROUTE_TABLE_ID}" --destination-cidr-block "${AWS_ACCOUNT_B_VPC_CIDR}" --vpc-peering-connection-id "${PEER_REQUEST_ID}"
+    logger "INFO" "Created route for ${SUBNET} to peering-connection in account A"
+  fi
+done
+
+switch_aws_credentials shared
+for SUBNET in $AWS_ACCOUNT_B_PRIVATE_SUBNET_IDS; do 
+  ROUTE_TABLE_ID=$(aws ec2 describe-route-tables --filters "Name=association.subnet-id,Values=${SUBNET}" --query 'RouteTables[*].RouteTableId' | jq -r '.[0]')
+  aws ec2 create-route --route-table-id "${ROUTE_TABLE_ID}" --destination-cidr-block "${AWS_ACCOUNT_A_VPC_CIDR}" --vpc-peering-connection-id "${PEER_REQUEST_ID}"
+  logger "INFO" "Created route for ${SUBNET} to peering-connection in account B"
+done
+
+
+# STEP. Create an IAM role in Account B which has a trust relationship with Account A(assume-role-policy) and 
+# add an EFS policy with necessary permissions, then attach the policy to the role.
+cat <<EOF > "${SHARED_DIR}"/AssumeRolePolicyInAccountB.json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "arn:aws:iam::${AWS_ACCOUNT_A_ID}:root"
+            },
+            "Action": "sts:AssumeRole"
+        }
+    ]
+}
+EOF
+
+
+
+ACCOUNT_B_ROLE_ARN=$(aws iam create-role \
+  --role-name "${CLUSTER_NAME}-cross-account-aws-efs-csi-operator" \
```

Comment:
```suggestion
  --role-name "${ACCOUNT_B_ROLE_NAME}" \
```

There's already a variable with this name defined above.

Reply:
Good catch, thank you, updated!